hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/Segment.java

```diff
@@ -240,22 +253,23 @@ protected CellComparator getComparator() {
 
   protected long internalAdd(Cell cell) {
     boolean succ = getCellSet().add(cell);
-    long s = AbstractMemStore.heapSizeChange(cell, succ);
-    updateMetaInfo(cell, s);
+    long s = updateMetaInfo(cell, succ, false);
     return s;
   }
 
-  protected void updateMetaInfo(Cell toAdd, long s) {
-    getTimeRangeTracker().includeTimestamp(toAdd);
+  protected long updateMetaInfo(Cell cellToAdd, boolean succ, boolean flattenning) {
+    long s = flattenning ? constantCellMetaDataSize : heapSizeChange(cellToAdd, succ);
```

Comment:
Looks like downcasting, but not sure ... Does Segment really need to know about flattening? Creational pattern?